src/table/table-role/grid-navigation.tsx

```diff
@@ -88,16 +89,18 @@ class GridNavigationProcessor {
     this._table = table;
 
     this.table.addEventListener('focusin', this.onFocusin);
-    this.table.addEventListener('focusout', this.onFocusout);
     this.table.addEventListener('keydown', this.onKeydown);
 
     this.updateFocusTarget();
 
+    const tableNodesObserver = new MutationObserver(this.onTableNodeMutation);
+    tableNodesObserver.observe(table, { childList: true, subtree: true });
```

Comment:
Can we do a more targeted observation instead of listening to updates in the entire table? using MutationObserver can cost us performance

Reply:
also, could we replace `focusout` with `blur` and avoid MutationObserver altogether?

Reply:
Thanks, there was actually a better way. Since all focusables are registered explicitly we can use the unregister events to capture node removals. I updated my approach so no observer now.